src/patients/related-persons/RelatedPersonTab.tsx

```diff
@@ -88,14 +88,17 @@ const RelatedPersonTab = (props: Props) => {
         <div className="col-md-12">
           <Panel title={t('patient.relatedPersons.label')} color="primary" collapsible>
             {relatedPersons ? (
-              <List>
-                {relatedPersons.map((r) => (
-                  <ListItem key={r.id}>{r.fullName}</ListItem>
-                ))}
-              </List>
-            ) : (
-              <h1>Loading...</h1>
```

Comment:
We still should display loading if the page is actually loading. 

I think we can add a `isLoading` state and set it to `true` right before fetching the related persons and then false when its finished`

Reply:
The loading message should still be displayed if the page is loading:

"Loading..." is displayed when `relatedPersons` is undefined, which it initially is when the page starts loading.

The bug was to do with the `setRelatedPersons()` call being inside `if (patient.relatedPersons)`, resulting in `relatedPersons` being undefined instead of empty if a patient has no related persons.

My commit has moved the `setRelatedPersons` call so that it is called even when a patient has no related persons (setting `relatedPersons` to an empty array).

Hopefully this solution is okay. If needed I could commit a different solution using a new `isLoading` state.

Reply:
Yeah, I think your changes definitely fixed the bug! I think we now need to add in the intended functionality of displaying a loading message while the page is loading, a warning message when there are no related persons, and the related persons list when there are related persons.

Reply:
I believe my changes already display the loading message whilst loading, the warning message when there are no related persons or the related persons list when there are related persons?

Reply:
You're right! My bad, I saw the deleted line in the diff and then didn't see the addition in a different spot!

I think the only thing to change is instead of the `Loading...` text, we could use the `Spinner` component

Reply:
I've replaced the `Loading...` text with the `Spinner` component in a new commit. :ok_hand: